### PR TITLE
Docs: add annotations reference doc

### PR DESCRIPTION
**Pull request by devholic**
Let users know how to configure annotations for the Application.

Refs https://github.com/oam-dev/kubevela/pull/3434

Signed-off-by: Sunghoon Kang <hoon@linecorp.com>

## Notes

Should not be merged before https://github.com/oam-dev/kubevela/pull/3434 got merged

### CODE REVIEW

File: docs/end-user/annotations.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+---
+title: Annotations
```
Comment (wonderflow) [inline docs/end-user/annotations.md]:
```suggestion
title: Service Account Integration
```

Comment (wonderflow) [reply docs/end-user/annotations.md]: Annotation may be not clear enough for this feature and users can hardly find the docs.

I think we can just clarify the service account mechanism here and tell users how to use it. Give some examples.

Comment (devholic) [reply docs/end-user/annotations.md]: Thanks for the feedback. I added an example and background to the document. Could you re-review?

**Review by wonderflow — APPROVED** [review]
great

**Pull request merged**